runtime/src/qio/qio.c

```diff
@@ -4388,6 +4388,16 @@ qioerr qio_get_fs_type(qio_file_t* fl, int* out)
   return 0;
 }
 
+qioerr qio_get_fd(qio_file_t* fl, int* out)
+{
+  if(fl != NULL && fl->fd != -1)
+    *out = fl->fd;
+  else
+    QIO_RETURN_CONSTANT_ERROR(ENOSYS, "no fd");
```

Comment:
Now that I look at it, it'd be nice if this error case set `*out = -1` just so the behavior is more predictable if the error is not checked.

Reply:
certainly 😄 made the changes for setting `out`